source/System.cpp

```diff
@@ -213,6 +215,8 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
 			child.PrintTrace("Error: Expected key to have a value:");
 			continue;
 		}
+		else if(!remove && child.Size() == valueIndex + 1 && key == "linger time")
+			requestedLingerTime = child.Value(valueIndex);
```

Comment:
The case of `remove "linger time"` should have been handled by this point.
Similarly, it's already ensured that this node has a value. Just skipping it entirely if there are extraneous tokens after that isn't something we normally do.
```suggestion
		else if(key == "linger time")
			requestedLingerTime = child.Value(valueIndex);
```

Reply:
The case of `remove (anything)` isn't handled up to that point. That giant "else if" chain handles adding and removing. Only the case of removing everything has been handled.

Checking the child size handles the special case of `add "linger time"` without a value.

I made it look nicer. How do you like it now?